cmd/controller-manager/app/controller_manager.go

```diff
@@ -303,7 +313,7 @@ func StartControllers(s *options.ControllerManagerServer,
 		glog.V(1).Info("Starting shared informers")
 		informerFactory.Start(stop)
 	} else {
-		glog.V(1).Infof("Skipping starting service-catalog controller because servicecatalog/v1alpha1 is not available")
+		return fmt.Errorf("unable to start service-catalog controller: servicecatalog/v1alpha1 is not available")
```

Comment:
Yeah, it'll cause a death and restart.  We're a Kubernetes-native application -- let's rely on Kubernetes health checks and auto-backoff to take care of this for us instead of trying to write our own logic.  Since we only have one API we care about, it's not like we want to have the potential to run some things while waiting for others.